src/linker/Linker.Dataflow/Scanner.cs

```diff
@@ -0,0 +1,984 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using ILLink.Shared;
+using ILLink.Shared.DataFlow;
+using ILLink.Shared.TrimAnalysis;
+using ILLink.Shared.TypeSystemProxy;
+using Mono.Cecil;
+using Mono.Cecil.Cil;
+using Mono.Collections.Generic;
+using static Mono.Linker.ParameterHelpers;
+using MultiValue = ILLink.Shared.DataFlow.ValueSet<ILLink.Shared.DataFlow.SingleValue>;
+
+namespace Mono.Linker.Dataflow
+{
+	abstract class Scanner
```

Comment:
Maybe a better name would be `LocalDataFlowScanner`:
* More descriptive (it really only handle local data flow - let's not count hoisted locals)
* It is a direct counterpart of `LocalDataFlowVisitor` from the analyzer.

Reply:
Basically I'm advocating for moving away from `MethodBodyScanner` and `ReflectionMethodBodyScanner` and embrace the terminology we have in analyzer and shared project.

That said, maybe this is not the right PR for such a change.

@sbomer, @tlakollo for opinions.

Reply:
Sounds good - maybe we should do it once we switch everything to the new CFG based scanner - so that we can only have TrimAnalysisScanner / TrimAnalysisHandler.